File: Prex.Utils/Prex.Utils/Entities/clsItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Prex.Utils.Entities
{
	public class clsItem
	{
		public object Valor { get; set; }
		public string Nombre { get; set; }
		public string Periodo { get; set; }
		public bool Format { get; set; }
		public string Tag { get; set; }


		public clsItem(object valor, string nombre)
		{
			Valor = valor;
			Nombre = nombre;
		}

		public clsItem(object valor, string nombre, string periodo, string tag) : this(valor, nombre)
		{
			Periodo = periodo;
			Tag = tag;
		}

		public clsItem(object valor, string nombre, bool format) : this(valor, nombre)
		{
			Format = format;
		}

		public override string ToString()
		{
			if (Format && DateTime.TryParse(Valor.ToString(), out DateTime d)) return d.ToString("dd/MM/yyyy");
			return Nombre;
		}
	}
}

File: VBP04000/VBP04000/Clases/clsItem.vb
Public Class clsItem

   Public Structure Item
      Dim Valor As Object
      Dim Nombre As String
		Dim Tag As String
		Dim Format As Boolean


		Public Sub New(ByVal _Valor As Object, ByVal _Nombre As String)
			Valor = _Valor
			Nombre = _Nombre
		End Sub

		Public Sub New(ByVal _Valor As Object, ByVal _Nombre As String, ByVal _Tag As String, ByVal formatearFecha As Boolean)
			Valor = _Valor
			Nombre = _Nombre
			Tag = _Tag
			Format = formatearFecha
		End Sub

		Public Overrides Function ToString() As String
			Dim d As DateTime = Nothing
			If Format AndAlso DateTime.TryParse(Valor, d) Then Return d.ToString("dd/MM/yyyy")
			Return Me.Nombre
		End Function

	End Structure

End Class
